app/models/message.js

```diff
@@ -0,0 +1,38 @@
+import Ember from 'ember';
+import DS from 'ember-data';
+const { getOwner } = Ember;
+
+var attr = DS.attr,
+belongsTo = DS.belongsTo;
+
+export default DS.Model.extend({
+
+  body: attr('string'),
+  isPrivate: attr('boolean'),
+  createdAt: attr('date'),
+  updatedAt: attr('date'),
+  state: attr('string', {
+    defaultValue: 'read'
+  }),
+  sender: belongsTo('user', {
+    async: true
+  }),
+  designation: belongsTo('designation', { async: false }),
+
+  myMessage: Ember.computed(function () {
+    var session = getOwner(this).lookup("service:session");
```

Comment:
Couldn't the service simply be added to the Model ? 
e.g 
```
export default DS.Model.extend({
    session: Ember.inject.service()
    //...
})
```
and then use `this.get('session')`

I'm no  Ember expert, but `getOwner(this).lookup` **_feels_** a bit hackish 🤔

Reply:
`session` service should be accessible everywhere as we've inject it using initializers

Reply:
@Darshan-Chauhan meaning we can do `this.get('session')` without injecting it ?

Reply:
@patrixr , yes. Look at this initializer => https://github.com/crossroads/stock.goodcity/blob/master/app/initializers/session.js

Reply:
aha, good to know :)

Reply:
yes, that would be right, thanks @Darshan-Chauhan @patrixr